src/bases2fastq/test.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+
+set -eou pipefail
+
+# Helper functions
+assert_file_exists() {
+  [ -f "$1" ] || { echo "File '$1' does not exist" && exit 1; }
+}
+
+assert_file_not_exists() {
+  [ ! -f "$1" ] || { echo "File '$1' does not exist" && exit 1; }
+}
+
+assert_directory_exists() {
+  [ -d "$1" ] || { echo "Directory '$1' does not exist" && exit 1; }
+}
+
+assert_file_not_empty() {
+  [ -s "$1" ] || { echo "File '$1' is empty but shouldn't be" && exit 1; }
+}
+assert_file_contains() {
+  grep -q "$2" "$1" || { echo "File '$1' does not contain '$2'" && exit 1; }
+}
+
+# Example output
+# Note that the format of the fastq file names and organization into subfolders
+# can differ based on the arguments provided to bases2fastq
+
+# |-- 20230404-Bases2Fastq-Sim_QC.html
+# |-- IndexAssignment.csv
+# |-- Metrics.csv
+# |-- RunManifest.csv
+# |-- RunManifest.json
+# |-- RunParameters.json
+# |-- RunStats.json
+# |-- Samples
```

Comment:
This seems incompatible with how the output of other demultiplexers is structured. This makes sense here, but maybe we could make a note of that in the description of the output arguments?

Reply:
I made sure that this component has similar output arguments to `bclconvert`: it also had `logs`, `reports`, and `output_directory` arguments? I'm also not sure what you mean with 'other demultiplexers', each vendor has their own output structure I think.